kv/memberlist/tcp_transport.go

```diff
@@ -376,24 +380,45 @@ func (t *TCPTransport) FinalAdvertiseAddr(ip string, port int) (net.IP, int, err
 				return nil, 0, fmt.Errorf("no private IP address found, and explicit IP not provided")
 			}
 
-			advertiseAddr = net.ParseIP(ip)
-			if advertiseAddr == nil {
-				return nil, 0, fmt.Errorf("failed to parse advertise address: %q", ip)
+			addr, err := netip.ParseAddr(ip)
+			if err != nil {
+				return nil, 0, fmt.Errorf("failed to parse advertise address %q", ip)
 			}
-		} else {
+
+			if addr.IsValid() {
```

Comment:
A couple of mutually exclusive questions:
1. If it's not valid, the execution will continue with the zero value `netip.Addr`. Is that correct? Shouldn't we return an error instead?
2. Can it be invalid? What's the case? Isn't `IsValid()` only returning false when it's the zero value?

Reply:
Yes good point.  I think I was just being defensive, but the error return above is all we care about.  I've added a commit to use the results directly and rely on `ParseAddr` to return an error when not valid.